specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-07-01/managedClusters.json

```diff
@@ -0,0 +1,5109 @@
+      "description": "A domain name that AKS agent nodes are reaching at.",
+      "type": "object",
+      "properties": {
+        "domainName": {
+          "description": "The domain name of the dependency.",
+          "type": "string"
+        },
+        "endpointDetails": {
+          "description": "The Ports and Protocols used when connecting to domainName.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDetail"
+          }
+        }
+      }
+    },
+    "EndpointDetail": {
+      "description": "connect information from the AKS agent nodes to a single endpoint.",
+      "type": "object",
+      "properties": {
+        "ipAddress": {
+          "description": "An IP Address that Domain Name currently resolves to.",
+          "type": "string"
+        },
+        "port": {
+          "format": "int32",
+          "description": "The port an endpoint is connected to.",
+          "type": "integer"
+        },
+        "protocol": {
+          "description": "The protocol used for connection",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description of the detail",
+          "type": "string"
+        }
+      }
+    },
+    "OutboundEnvironmentEndpoint": {
+      "description": "Egress endpoints which AKS agent nodes connect to for common purpose.",
+      "type": "object",
+      "properties": {
+        "category": {
+          "description": "The category of endpoints accessed by the AKS agent node, e.g. azure-resource-management, apiserver, etc.",
+          "type": "string"
+        },
+        "endpoints": {
+          "description": "The endpoints that AKS agent nodes connect to",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDependency"
+          }
+        }
+      }
+    },
+    "OutboundEnvironmentEndpointCollection": {
+      "description": "Collection of OutboundEnvironmentEndpoint",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OutboundEnvironmentEndpoint"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ManagedClusterSecurityProfile": {
+      "type": "object",
+      "properties": {
+        "azureDefender": {
+          "$ref": "#/definitions/ManagedClusterSecurityProfileAzureDefender",
+          "description": "Azure Defender settings for the security profile."
+        }
+      },
+      "description": "Security profile for the container service cluster."
+    },
+    "ManagedClusterSecurityProfileAzureDefender": {
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "type": "boolean",
```

Comment:
Please consider using a string enum instead of boolean.

**Replace boolean/switch properties with better enum**
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html